### ISSUE TITLE
[DynamicPage]: Screen reader is incorrectly announcing the name, role and state information for the non-interactive header section.

### ISSUE THREAD

**Issue by raoakash01** [SAP/ui5-webcomponents#12053]
### Describe the bug

We get announcements like "Toggle Header" even if there is no use case of toggle Header
I have attached the example of latest version and I have tested with the version we are using there it also announces Expanded Header.
adding screen shots for reference 

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/528df756-8f1e-44dd-8218-8eaa52b5527d" />
<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/963f4b99-9b8e-47b6-8eee-be99b98f5e8c" />
<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/9534cdd8-02ad-4846-97be-398d10d34b47" />

### Isolated Example

https://stackblitz.com/edit/github-3javgvfp?file=src%2FApp.tsx

### Reproduction steps

For windows -> Enable JAWS and navigate the page content using down arrow icon in VPC mode.
For Mac -> use voice over and navigate with control+option+side arrow key

### Expected Behaviour

Ensure that screen readers do not announce the unnecessary name, role, or state information for non-interactive header sections, as this may cause confusion for users.

### UI5 Web Components for React Version

2.7.0

### UI5 Web Components Version

2.7.0

### Browser

Chrome

### Relevant log output

```shell

```

### Declaration

- [x] I’m not disclosing any internal or sensitive information.

**Comment by Lukas742** [comment]
Thanks for reporting! I'll forward this issue to our UI5 Web Components Colleagues as the affected component is developed in their repository.

**Comment by kgogov** [comment]
Hello @SAP/ui5-webcomponents-topic-p,

Could you please take a look at the following issue?

It seems that when we have a DynamicPageTitle without the snapping functionality enabled, JAWS incorrectly announces “Toggle Header,” which is misleading because the header cannot actually be toggled.

Regards,
Konstantin

![Image](https://github.com/user-attachments/assets/d4a54a86-32ee-4821-89ee-1bae5caf4356)

![Image](https://github.com/user-attachments/assets/4210c029-2a4e-4489-b103-ac5daa9307c9)
